### DISCUSSION TITLE
Docs not up-to-date on conventional commits specification?

### DISCUSSION THREAD

**Discussion by jmartens** [GitTools/GitVersion#4326] [Q&A]
I was looking to enable GitVersion for my repository where I use conventional style commit message.

Your [docs](https://gitversion.net/docs/reference/version-increments#conventional-commit-messages) states that we should use the following configuration:
```
mode: MainLine # Only add this if you want every version to be created automatically on your main branch.
major-version-bump-message: "^(build|chore|ci|docs|feat|fix|perf|refactor|revert|style|test)(\\([\\w\\s-,/\\\\]*\\))?(!:|:.*\\n\\n((.+\\n)+\\n)?BREAKING CHANGE:\\s.+)"
minor-version-bump-message: "^(feat)(\\([\\w\\s-,/\\\\]*\\))?:"
patch-version-bump-message: "^(fix|perf)(\\([\\w\\s-,/\\\\]*\\))?:"
```

However when I do so I get the following error when running GitVersion on the command line I get the following error:

> Could not build the configuration instance because following exception occurred: 'Requested value 'MainLine' was not found.' Please ensure that the /overrideconfig parameters are correct and the configuration file is in the correct format.

<details><summary>Full console output</summary>

```
$ gitversion
INFO [24-11-21 13:17:10:78] Applicable build agent found: 'LocalBuild'.
INFO [24-11-21 13:17:10:82] Working directory: C:\Users\jonathan.martens\Code\renovate-config
INFO [24-11-21 13:17:10:86] Project root is: C:\Users\jonathan.martens\Code\renovate-config\
INFO [24-11-21 13:17:10:86] DotGit directory is: C:\Users\jonathan.martens\Code\renovate-config\.git
INFO [24-11-21 13:17:10:86] Branch from build environment:
INFO [24-11-21 13:17:10:87] Found configuration file at 'C:\Users\jonathan.martens\Code\renovate-config\GitVersion.yaml'
INFO [24-11-21 13:17:10:90] -< Begin: Loading version variables from disk cache file C:\Users\jonathan.martens\Code\renovate-config\.git\gitversion_cache\B5D977623B140B8CF5763802ABC143FEE1D3598E >-
  INFO [24-11-21 13:17:10:90] Cache file C:\Users\jonathan.martens\Code\renovate-config\.git\gitversion_cache\B5D977623B140B8CF5763802ABC143FEE1D3598E not found.
INFO [24-11-21 13:17:10:90] -< End: Loading version variables from disk cache file C:\Users\jonathan.martens\Code\renovate-config\.git\gitversion_cache\B5D977623B140B8CF5763802ABC143FEE1D3598E (Took: 0.46ms) >-
INFO [24-11-21 13:17:10:91] Found configuration file at 'C:\Users\jonathan.martens\Code\renovate-config\GitVersion.yaml'
WARN [24-11-21 13:17:11:10] An error occurred:
Could not build the configuration instance because following exception occurred: 'Requested value 'MainLine' was not found.' Please ensure that the /overrideconfig parameters are correct and the configuration file is in the correct format.
$
```

</details>

**Comment by jmartens** [comment]
Hmmm, just found this in https://github.com/GitTools/GitVersion/blob/main/BREAKING_CHANGES.md

> The Mainline mode and the related implementation has been removed completely. The new Mainline version strategy should be used instead.

What is the current default config then?

**Reply by HHobeck** [reply]
Please try the following `TrunkBased` workflow configuration (gitversion.yaml):

```yaml
workflow: TrunkBased/preview1
```

May I ask you to create a pull-request to update the documentation?
- https://github.com/GitTools/GitVersion/issues/4258

Is this really a valid regular expression?

```yaml
major-version-bump-message: "^(build|chore|ci|docs|feat|fix|perf|refactor|revert|style|test)(\\([\\w\\s-,/\\\\]*\\))?(!:|:.*\\n\\n((.+\\n)+\\n)?BREAKING CHANGE:\\s.+)"
minor-version-bump-message: "^(feat)(\\([\\w\\s-,/\\\\]*\\))?:"
patch-version-bump-message: "^(fix|perf)(\\([\\w\\s-,/\\\\]*\\))?:"
```

The workflow templates can be found at: [Reference - Configuration](https://gitversion.net/docs/reference/configuration)

**Comment by jmartens** [comment]
> Please try the following TrunkBased workflow configuration (gitversion.yaml):
workflow: 

```
TrunkBased/preview1
```

> May I ask you to create a pull-request to update the documentation?

I will see what I can do. 

> Is this really a valid regular expression?
```
major-version-bump-message: 
"^(build|chore|ci|docs|feat|fix|perf|refactor|revert|style|test)(\\([\\w\\s-,/\\\\]*\\))?(!:|:.*\\n\\n((.+\\n)+\\n)?BREAKING CHANGE:\\s.+)"
minor-version-bump-message: "^(feat)(\\([\\w\\s-,/\\\\]*\\))?:"
patch-version-bump-message: "^(fix|perf)(\\([\\w\\s-,/\\\\]*\\))?:"
```

It was copied verbatim from your docs. It looks pretty ok to me.

> The the supported workflows can be found at: Reference - Configuration

Thanks, will study it some more, perhaps during the weekend.

**Reply by HHobeck** [reply]
https://regex101.com/r/3bLeiC/1

![image](https://github.com/user-attachments/assets/038fe16c-81b3-49d2-9304-32b143aa323f)

Aside from the fact that it's even valid, this regex doesn't make any sense to me either. For instance `\\w` regex matches on the one hand to a backslash and the letter `w` where on the other hand the `\w` (which might be intended?) matches to any word character. For me it looks like that the regex string was just copied from c# where the special characters have been escaped.

@jmartens: Do you know the specification for conventional commits?

**Reply by jmartens** [reply]
> @jmartens: Do you know the specification for conventional commits?

https://www.conventionalcommits.org/en/v1.0.0/
